graphene/types/field.py

```diff
@@ -89,6 +89,9 @@ def __init__(
         ), f'The default value can not be a function but received "{base_type(default_value)}".'
 
         if required:
+            assert (
+                deprecation_reason is None
+            ), f"Field {name} is required, cannot deprecate it."
```

Comment:
This restriction only applies to _input object fields_ & arguments : https://spec.graphql.org/draft/#sec--deprecated